stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
+    def add_transition(self, name: str, transition: tuple[Pattern[str], str, str]) -> None: ...
+    def remove_transition(self, name: str) -> None: ...
+    def make_transition(self, name: str, next_state: str | None = None) -> tuple[Pattern[str], _TransitionMethod, str]: ...
+    def make_transitions(
+        self, name_list: list[str | tuple[str] | tuple[str, str]]
+    ) -> tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]: ...
+    def no_match(
+        self, context: list[str], transitions: tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]
+    ) -> _TransitionResult: ...
+    def bof(self, context: list[str]) -> tuple[list[str], list[str]]: ...
+    def eof(self, context: list[str]) -> list[str]: ...
+    def nop(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class StateMachineWS(StateMachine):
+    def get_indented(self, until_blank: bool = False, strip_indent: bool = True) -> tuple[StringList, int, int, bool]: ...
+    def get_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True
+    ) -> tuple[list[str], int, bool]: ...
+    def get_first_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True, strip_top: bool = True
+    ) -> tuple[list[str], int, int, bool]: ...
+
+class StateWS(State):
+    indent_sm: type[StateMachine] | None
+    indent_sm_kwargs: dict[str, Any] | None
+    known_indent_sm: type[StateMachine] | None
+    known_indent_sm_kwargs: dict[str, Any] | None
+    ws_patterns: dict[str, str]
+    ws_initial_transitions: Sequence[str]
+    def __init__(self, state_machine: StateMachine, debug: bool = False) -> None: ...
+    def add_initial_transitions(self) -> None: ...
+    def blank(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def known_indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def first_known_indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class _SearchOverride:
+    def match(self, pattern: Pattern[str]) -> Match[str]: ...
+
+class SearchStateMachine(_SearchOverride, StateMachine): ...
+class SearchStateMachineWS(_SearchOverride, StateMachineWS): ...
+
+_Data = TypeVar("_Data")
+
+class ViewList(Generic[_Data]):
+    data: list[_Data]
+    items: list[tuple[str, int]]
+    parent: ViewList[_Data]
+    parent_offset: int
+    def __init__(
+        self,
+        initlist: ViewList[_Data] | list[_Data] | None = None,
+        source: str | None = None,
+        items: list[tuple[str, int]] | None = None,
+        parent: ViewList[_Data] | None = None,
+        parent_offset: int | None = None,
+    ) -> None: ...
+    def __lt__(self, other: Any) -> bool: ...
```

Comment:
> I think you should use __other so that it indicates (without enforcing) that you expect positional only arguments. I don't know whether you can use positional-only arguments syntax or not.

i haven't noticed that pattern elsewhere in typeshed. I'd like to follow convention here. @JelleZijlstra?

Reply:
We should use PEP 570 pos-only syntax in new code, not double underscores. (We had some technical reasons why we didn't migrate all existing code yet.) Here I would make the parameter positional-only if it's also positional-only at runtime. If the class is implemented in Python, it probably isn't.